From 19b284cb4649e514a9feedb9a3c64a5ee485b721 Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Thu, 19 Sep 2024 18:26:03 +0800
Subject: [PATCH 214/277] mtk: mt76: mt7996: remove unnecessary failed log

These failed logs are not a problem, since there might be many clients
doing connection and disconnection.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/mac.c | 10 +++-------
 mt7996/mcu.c | 13 -------------
 2 files changed, 3 insertions(+), 20 deletions(-)

diff --git a/mt7996/mac.c b/mt7996/mac.c
index 86dc32e1..851b753c 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -2887,17 +2887,13 @@ static int mt7996_mac_sta_poll(struct mt76_dev *dev)
 
 	ret = mt7996_mcu_get_per_sta_info(dev, UNI_PER_STA_RSSI, i, sta_list);
 	if (ret)
-		dev_err(dev->dev, "Failed to update RSSI of polled STAs.\n");
+		return ret;
 
 	ret = mt7996_mcu_get_per_sta_info(dev, UNI_PER_STA_SNR, i, sta_list);
 	if (ret)
-		dev_err(dev->dev, "Failed to update SNR of polled STAs.\n");
+		return ret;
 
-	ret = mt7996_mcu_get_per_sta_info(dev, UNI_PER_STA_PKT_CNT, i, sta_list);
-	if (ret)
-		dev_err(dev->dev, "Failed to update MSDU counts of polled STAs.\n");
-
-	return ret;
+	return mt7996_mcu_get_per_sta_info(dev, UNI_PER_STA_PKT_CNT, i, sta_list);
 }
 
 void mt7996_mac_work(struct work_struct *work)
diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index f6587a66..3c873ed7 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -6704,10 +6704,6 @@ int mt7996_mcu_get_per_sta_info(struct mt76_dev *dev, u16 tag,
 				mlink->ack_signal = mt76_rx_signal(phy->antenna_mask,
 								   mlink->chain_ack_signal);
 				ewma_avg_signal_add(&mlink->avg_ack_signal, -mlink->ack_signal);
-			} else {
-				ret = -EINVAL;
-				dev_err(dev->dev, "Failed to update RSSI for "
-				                  "invalid WCID: %hu\n", wlan_idx);
 			}
 		}
 		break;
@@ -6719,11 +6715,6 @@ int mt7996_mcu_get_per_sta_info(struct mt76_dev *dev, u16 tag,
 			if (mlink)
 				memcpy(mlink->chain_ack_snr, res->snr[i].val,
 				       IEEE80211_MAX_CHAINS);
-			else {
-				ret = -EINVAL;
-				dev_err(dev->dev, "Failed to update SNR for "
-				                  "invalid WCID: %hu\n", wlan_idx);
-			}
 		}
 		break;
 	case UNI_PER_STA_PKT_CNT:
@@ -6736,10 +6727,6 @@ int mt7996_mcu_get_per_sta_info(struct mt76_dev *dev, u16 tag,
 
 				wcid->stats.tx_packets_retried += retries;
 				wcid->stats.tx_packets_failed += retries + drops;
-			} else {
-				ret = -EINVAL;
-				dev_err(dev->dev, "Failed to update MSDU counts for "
-						  "invalid WCID: %hu\n", wlan_idx);
 			}
 		}
 		break;
-- 
2.45.2

